File: Clue/CardType.py
#!/usr/bin/python

class CardType:
    
    cardType = None
    character = 0
    weapon = 1
    room = 2
    types = ["CHARACTER", "WEAPON", "ROOM"]

    def __init__(self, cardType):

        index = None

        #check for numeric input then string input
        try:
            index = int(userInput)
        except ValueError:
            index = getCardType(cardType)

        if index > 0 and index < 3:
                self.cardType = types.get(index)
                return True
        else:
            return False

    @staticmethod
    def getCardType(x):

        return {
            'CHARACTER': 0,
            'WEAPON': 1,
            "ROOM": 2,
            }.get(x, -1)
    
    @property
    def character(self):
        return type(self).character
    @property
    def weapon(self):
        return type(self).weapon
    @property
    def room(self):
        return type(self).room

File: Java/src/clue/logic/CardType.java
package clue.logic;

public enum CardType {
    ROOM("ROOM", 0), 
    WEAPON("WEAPON", 1), 
    CHARACTER("CHARACTER", 2), 
    UNKNOWN("UNKNOWN", 3);

    private final String label;
    private final int index;

    CardType(String str, int i) {
        label = str;
        index = i;
    }

    public String toString() {
        return this.label;
    }
}